lib/kubeclient/watch_stream.rb

```diff
@@ -21,16 +21,27 @@ def each
         end
 
         buffer = ''
+        done = false
         response.body.each do |chunk|
+          done = false
           buffer << chunk
           while (line = buffer.slice!(/.+\n/))
             yield(@format == :json ? WatchNotice.new(JSON.parse(line)) : line.chomp)
           end
+          done = true
         end
+        done
```

Comment:
the only way I see this can be false is if connection was interrupted before at least 1 complete line was received?